File: Tracko-Flutter-UI/lib/models/split.dart
import 'package:tracko/models/transaction.dart';
import 'package:tracko/models/user.dart';
import 'package:tracko/models/contact.dart';

class Split {
  int? id;
  int transactionId = 0;
  int userId = 0;
  int? contactId;
  double amount = 0.0;
  int isSettled = 0;
  DateTime settledAt = DateTime.now();
  Transaction? transaction;
  Contact? contact;

  Split();

  @override
  String toString() {
    return 'Split{id: $id, transactionId: $transactionId, userId: $userId, contactId: $contactId, amount: $amount, isSettled: $isSettled}';
  }
}

File: backend/src/main/java/com/trako/entities/Split.java
package com.trako.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name = "splits")
@Getter
@Setter
public class Split {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @NotNull
    @Column(name = "transaction_id")
    private Long transactionId;

    @NotNull
    @Column(name = "user_id")
    private String userId;

    @NotNull
    @jakarta.validation.constraints.Positive
    @Column(name = "amount")
    private Double amount;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "settled_at")
    private Date settledAt;

    @Column(name = "is_settled")
    private Integer isSettled = 0;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "transaction_id", insertable = false, updatable = false)
    private Transaction transaction;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @Column(name = "contact_id")
    private Long contactId;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "contact_id", insertable = false, updatable = false)
    private Contact contact;
}
